src/AEG/AEGTree.ts

```diff
@@ -12,13 +12,45 @@ export class AEGTree {
     }
 
     /**
-     * Recursively ensures the bounding box structure of this AEG is consistent with
-     * the hierarchy. It is consistent if bounding boxes of children nodes are within the bounding
-     * box of the parent node.
-     * @returns True, if the structure is consistent. Else, false.
+     * Calls verifyAEG() with the sheet of assertion as its argument.
+     * @returns the result of verifyAEG() called with the sheet of assertion
      */
-    public verifyAEG(): boolean {
-        return this.sheet.verifyCut();
+    public verify(): boolean {
+        return this.verifyAEG(this.sheet);
+    }
+
+    /**
+     * Checks for consistency in the tree's structure.
+     * Structural consistency is achieved when:
+     * All bounding boxes of currentCut's child nodes are within the bounding box of currentCut,
+     * The same is true for each CutNode within currentCut,
+     * And the children of each cut level do not overlap with each other.
+     * @param currentCut: The cut for which we are checking consistency.
+     * @returns True, if the structure is structurally consistent. Else, false.
+     */
+    private verifyAEG(currentCut: CutNode): boolean {
+        for (let i = 0; i < currentCut.children.length; i++) {
+            //Check that all children, in this level, are in currentCut
+            if (!currentCut.containsNode(currentCut.children[i])) {
+                return false;
+            }
+
+            //Check one level deeper if the child is a CutNode
+            if (
+                currentCut.children[i] instanceof CutNode &&
+                !this.verifyAEG(currentCut.children[i] as CutNode)
+            ) {
+                return false;
+            }
+
+            //Check for overlaps on the same level
+            for (let j = 0; j < currentCut.children.length; j++) {
+                if (this.overlaps(currentCut.children[i], currentCut.children[j])) {
+                    return false;
+                }
+            }
```

Comment:
This will check if a child overlaps with itself, returning false every time. additionally, it's extremely inefficient since it checks overlaps of objects twice overlaps is a symmetric relation, i.e., `overlaps(a,b) == overlaps(b,a)` hence you only need to check one of them. You can *probably* fix both of these issues by only checking the upper triangular of the pairing matrix this setup generates, by starting the second loop at `i+1` instead of 0
```suggestion
            for (let j = i+1; j < currentCut.children.length; j++) {
                if (this.overlaps(currentCut.children[i], currentCut.children[j])) {
                    return false;
                }
            }
```